src/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -79,5 +78,211 @@ public static void ThrownFromReader()
+        {
+            try
+            {
+                JsonSerializer.Parse<List<int>>(@"[1, bad]");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$[1]", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathFor2dArrayFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<int[][]>(@"[[1, 2],[3,bad]]");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$[1][1]", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathFor2dListFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<List<List<int>>>(@"[[1, 2],[3,bad]]");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$[1][1]", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathForChildPropertyFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<RootClass>(@"{""ChildClass"":{""MyInt"":bad]}");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$.ChildClass.MyInt", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathForChildListFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<RootClass>(@"{""ChildClass"":{""MyIntArray"":[1, bad]}");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$.ChildClass.MyIntArray[1]", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathForChildDictionaryFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<RootClass>(@"{""ChildClass"":{""MyIntDictionary"":{""Key"": bad]");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$.ChildClass.MyIntDictionary.Key", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathForChildDictionaryWithSpecialCharacterFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<RootClass>(@"{""ChildClass"":{""MyIntDictionary"":{""K.e.y"": bad]");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$.ChildClass.MyIntDictionary['K.e.y']", e.Path);
```

Comment:
I'd like some tests that have multiple escaped keys mixed with arrays. Make sure no rogue `.` appear in the path.

Reply:
Sure. Adding a couple - if I follow, an array should not have a `.` before the `[`